src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,310 @@
+import {IRocketPoolDeposit} from './interfaces/IRocketPoolDeposit.sol';
+
+contract AaveCurator is VersionedInitializable {
+  using SafeERC20 for IERC20;
+
+  event AdminChanged(address indexed oldAdmin, address indexed newAdmin);
+  event DepositedIntoV2(address indexed token, uint256 amount);
+  event DepositedIntoV3(address indexed token, uint256 amount);
+  event ManagerChanged(address indexed oldAdmin, address indexed newAdmin);
+  event SwapCanceled(address fromToken, address toToken, uint256 amount);
+  event SwapRequested(address fromToken, address toToken, uint256 amount);
+  event TokenUpdated(address indexed token, bool allowed);
+
+  error Invalid0xAddress();
+  error InvalidAmount();
+  error InvalidCaller();
+  error InvalidOracleType();
+  error InvalidRecipient();
+  error InvalidToken();
+  error OracleNotSet();
+
+  enum TokenType {
+    Standard,
+    BPT8020
+  }
+
+  struct TokenChailinkOracleData {
+    address oracle;
+    bool isEthBased;
+  }
+
+  modifier onlyAdmin() {
+    if (msg.sender != admin) revert InvalidCaller();
+    _;
+  }
+
+  modifier onlyAdminOrManager() {
+    if (msg.sender != admin && msg.sender != manager) {
+      revert InvalidCaller();
+    }
+    _;
+  }
+
+  IRocketPoolDeposit private constant ROCKET_POOL =
+    IRocketPoolDeposit(0xDD3f50F8A6CafbE9b31a427582963f465E745AF8);
+  IWeth private constant WETH = IWeth(AaveV2EthereumAssets.WETH_UNDERLYING);
+
+  address public constant STETH = AaveV2EthereumAssets.stETH_UNDERLYING;
+  address public constant BPT_PRICE_CHECKER = 0x7961bBC81352F26d073aA795EED51290C350D404; // TODO: Update with new one
+  address public constant CHAINLINK_PRICE_CHECKER = 0x4D2c3773E69cB69963bFd376e538eC754409ACFa;
+
+  address public admin = AaveGovernanceV2.SHORT_EXECUTOR;
+  address public milkman = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public manager;
+
+  mapping(address tokenAddress => bool) public allowedFromTokens;
+  mapping(address tokenAddress => bool) public allowedToTokens;
+  mapping(address tokenAddress => TokenChailinkOracleData) public tokenChainlinkOracle;
+
+  /// @notice Current revision of the contract.
+  uint256 public constant REVISION = 1;
+
+  function initialize() external initializer {
+    admin = AaveGovernanceV2.SHORT_EXECUTOR;
+    emit AdminChanged(address(0), admin);
+  }
+
+  function swap(
+    address fromToken,
+    address toToken,
+    address recipient,
+    uint256 amount,
+    uint256 slippage,
+    TokenType _tokenType
+  ) external onlyAdminOrManager {
+    if (amount == 0) revert InvalidAmount();
+    if (!allowedFromTokens[fromToken]) revert InvalidToken();
+    if (!allowedToTokens[toToken]) revert InvalidToken();
+    if (recipient != address(this) && recipient != address(AaveV3Ethereum.COLLECTOR)) {
+      revert InvalidRecipient();
+    }
+
+    console2.log('before');
+    IERC20(fromToken).allowance(address(this), milkman);
+    IERC20(fromToken).approve(milkman, amount);
```

Comment:
I would use oz's SafeErc20 to handle tokens like USDT, where you would call safeIncreaseAllowance IIRC